core/blockchain/impl/level_db_block_tree.cpp

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "blockchain/impl/level_db_block_tree.hpp"
+
+#include <boost/assert.hpp>
+#include "blockchain/impl/common.hpp"
+#include "blockchain/impl/level_db_util.hpp"
+#include "common/visitor.hpp"
+#include "crypto/blake2/blake2b.h"
+#include "scale/scale.hpp"
+#include "storage/leveldb/leveldb_error.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::blockchain, LevelDbBlockTree::Error, e) {
+  using E = kagome::blockchain::LevelDbBlockTree::Error;
+  switch (e) {
+    case E::INVALID_DB:
+      return "genesis block is not provided, and the database is either empty "
+             "or does not contain valid block tree";
+    case E::HASH_FAILED:
+      return "attempt to hash block part has failed";
+    case E::NO_SUCH_BLOCK:
+      return "block with such hash cannot be found in the local storage";
+    case E::INTERNAL_ERROR:
```

Comment:
What is the difference between this and unknown error?

Reply:
Unknown error signalizes we forgot to handle some of the error cases. An internal error means that something bad happened, but we don't want to tell the client exactly what, as level of abstraction of that error is too low